### ISSUE TITLE
Syntax error. Escape "order" field.

### ISSUE THREAD

**Issue by arso8** [Raizlabs/DBFlow#335]
I have the same problem, like in #182  when insert row with **already existing** PK.
**DBFlow version 2.2.1**

```java
MyAboutImage myAboutImage = new MyAboutImage();
myAboutImage.set_id(1);
myAboutImage.setAbout(myAbout);
myAboutImage.setImage_id("");
myAboutImage.setOrder(1);
myAboutImage.save();
```
```
Caused by: android.database.sqlite.SQLiteException: near "order": syntax error (code 1): , while compiling: UPDATE  OR ABORT about_image SET order=?,image_id=?,date_update=?,date_recovery=?,date=?,date_drop=?,_id=?,type=?,about_id=? WHERE `_id`=1
            at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
            at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:891)
            at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:502)
            at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
            at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
            at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
            at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1574)
            at com.raizlabs.android.dbflow.SQLiteCompatibilityUtils.updateWithOnConflict(SQLiteCompatibilityUtils.java:63)
            at com.raizlabs.android.dbflow.sql.SqlUtils.update(SqlUtils.java:289)
            at com.raizlabs.android.dbflow.sql.SqlUtils.save(SqlUtils.java:264)
            at com.raizlabs.android.dbflow.structure.ModelAdapter.save(ModelAdapter.java:54)
            at com.raizlabs.android.dbflow.structure.BaseModel.save(BaseModel.java:50)
            at com.universal.test.utils.TestData.dbInsertTest(TestData.java:217)
            at com.universal.test.activities.MainActivity.onCreate(MainActivity.java:89)
            at android.app.Activity.performCreate(Activity.java:5990)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2309)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2418)
            at android.app.ActivityThread.access$900(ActivityThread.java:154)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5289)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
```
My model:
```java
@Table(databaseName = Constants.DB.DB_NAME, tableName = Constants.DB.TABLE.ABOUT_IMAGE)
public class MyAboutImage extends BaseModel {
    @Column
    @PrimaryKey
    int _id;
    @Column
    @ForeignKey(
            references = {@ForeignKeyReference(columnName = "about_id",
                    columnType = Integer.class,
                    foreignColumnName = "_id")},
            saveForeignKeyModel = false)
    MyAbout about;
    @Column
    String image_id;
    @Column
    int order;    //  <- here's the problem field
    @Column
    int date;
    @Column
    int date_update;
    @Column
    int date_drop;
    @Column
    int date_recovery;
    @Column
    int type;
}
```

**Comment by romulof** [comment]
Same problem here

**Comment by ydanneg** [comment]
https://github.com/Raizlabs/DBFlow/issues/178 related

**Comment by ydanneg** [comment]
http://www.sqlite.org/lang_keywords.html

**Comment by ydanneg** [comment]
The issue occures on UPDATING the model ONLY. (e.g. second save() call will fail on such model)
DBFlow updates model using ContentValues that are failed to compile by Android if key is a reserved word.

This could be fixed by wrapping ContentValues keys into quotes but this will cause lot of other problems on the other side - Content Providers.
It is very compicated to fix this issue using current design.

IMHO this issue can be closed with the resolution as "not to use SQLIte reserved words as column names":

```
Column(name = "order_id")
int order; 
```

PS.
I wonder why DBFlow uses grave accent(`) to quote SQL identifiiers. SQLite recommends (http://www.sqlite.org/lang_keywords.html) using double-quotes(") to escape identifiers and single-quotes(') to escape string literals. 
I would suggest thinking about ability to configure DBFlow with the default identifier and string literal escape chars.

**Comment by agrosner** [comment]
I will figure out a way to configure the quoting mechanism and probably deprecate the "`" since its not recommended. This requires a large change. I will table it for 3.0

**Comment by machei** [comment]
+1

**Comment by agrosner** [comment]
fixed in 3.0, only quote column names if needed.
